pkg/liqoctl/virtualnode/create.go

```diff
@@ -0,0 +1,214 @@
+			output.ExitOnErr(o.handleCreate(ctx))
+		},
+	}
+
+	cmd.Flags().VarP(outputFormat, "output", "o",
+		"Output the resulting VirtualNode resource, instead of applying it. Supported formats: json, yaml")
+
+	// TODO: check validity of both cluster-id and cluster-name
+	cmd.Flags().StringVar(&o.remoteClusterIdentity.ClusterID, "cluster-id", "", "The cluster ID of the remote cluster")
+	cmd.Flags().StringVar(&o.remoteClusterIdentity.ClusterName, "cluster-name", "", "The cluster name of the remote cluster")
+	cmd.Flags().BoolVar(&o.createVirtualNode, "create-node",
+		true, "Create a node to target the remote cluster (and schedule on it)")
+	cmd.Flags().StringVar(&o.kubeconfigSecretName, "kubeconfig-secret-name",
+		"", "The name of the secret containing the kubeconfig of the remote cluster")
+	cmd.Flags().StringVar(&o.cpu, "cpu", "2", "The amount of CPU available in the virtual node")
+	cmd.Flags().StringVar(&o.memory, "memory", "4Gi", "The amount of memory available in the virtual node")
+	cmd.Flags().StringVar(&o.pods, "pods", "110", "The amount of pods available in the virtual node")
+	cmd.Flags().StringSliceVar(&o.storageClasses, "storage-classes",
+		[]string{}, "The storage classes offered by the remote cluster. The first one will be used as default")
+	cmd.Flags().StringToStringVar(&o.labels, "labels", map[string]string{}, "The labels to be added to the virtual node")
+
+	runtime.Must(cmd.MarkFlagRequired("cluster-id"))
+	runtime.Must(cmd.MarkFlagRequired("cluster-name"))
+	runtime.Must(cmd.MarkFlagRequired("kubeconfig-secret-name"))
+
+	runtime.Must(cmd.RegisterFlagCompletionFunc("cluster-id", completion.ClusterIDs(ctx,
+		o.createOptions.Factory, completion.NoLimit)))
+	runtime.Must(cmd.RegisterFlagCompletionFunc("cluster-name", completion.ClusterNames(ctx,
+		o.createOptions.Factory, completion.NoLimit)))
+	runtime.Must(cmd.RegisterFlagCompletionFunc("kubeconfig-secret-name", completion.KubeconfigSecretNames(ctx,
+		o.createOptions.Factory, completion.NoLimit)))
+
+	return cmd
+}
+
+func (o *Options) handleCreate(ctx context.Context) error {
+	opts := o.createOptions
+	if opts.OutputFormat != "" {
+		opts.Printer.CheckErr(o.output())
+		return nil
+	}
+
+	s := opts.Printer.StartSpinner("Creating virtual node")
+
+	virtualNode := o.getVirtualNode()
+	_, err := controllerutil.CreateOrUpdate(ctx, opts.CRClient, virtualNode, func() error {
+		return o.mutateVirtualNode(virtualNode)
+	})
+	if err != nil {
+		s.Fail("Unable to create virtual node: %v", output.PrettyErr(err))
+		return err
+	}
+	s.Success("Virtual node created")
+
+	if virtualNode.Spec.CreateNode != nil && *virtualNode.Spec.CreateNode {
+		waiter := wait.NewWaiterFromFactory(opts.Factory)
+		// TODO: we cannot use the cluster identity here
+		if err := waiter.ForNode(ctx, virtualNode.Spec.ClusterIdentity); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *Options) getVirtualNode() *virtualkubeletv1alpha1.VirtualNode {
```

Comment:
```suggestion
func (o *Options) forgeVirtualNode() *virtualkubeletv1alpha1.VirtualNode {
```
Maybe it's more understandable? It seems it gets a VirtualNode which already exists from the name.